### ISSUE TITLE
Add possibility to render real DOM node instead of vnode

### ISSUE THREAD

**Issue by jhnns** [wavesoft/dot-dom#45]
Hey 👋 

awesome project 👍.

I think, this could be a really interesting library for Web Components that don't want to render their contents using the imperative DOM API but still want to stay as small as possible. For obvious reasons, Web Components can't just include their own rendering library (or do we want to end up with 10 rendering libraries in one application? 😁). But when the rendering library plus the rendering code inside the Web Component is actually smaller than code that just uses the verbose DOM API, things get interesting...

Anyway, what I was thinking: There are some edge cases where the dot-dom API is not flexible enough (e.g. registering event listeners in the capturing phase). What do think about adding the possibility to just render real DOM nodes in that case?

In the event listeners use-case:

```js
function Button({onCaptureClick}) {
    const button = document.createElement("button");

    button.addEventListener("click", onCaptureClick, true);

    return button;
}
```

A DOM node could appear everywhere where VNodes are allowed. This could be used as an escape hatch for all situations where the declarative API is not flexible enough.

I haven't thought this through, but judging from the source code, I think it might be doable within the `512` limit. Just wanted to ask you what you think about that before I spend time implementing that feature.

**Comment by wavesoft** [comment]
Oh that's an interesting idea @jhnns . I have a gut feeling that detecting differences between states might be a bit tricky, but apart from this it should be a simple task. If you manage to squeeze this within the 512 byte limit, it would be super awesome.

Note however that I am currently doing some refactoring on the [`ic/smarter-state-perservation`](https://github.com/wavesoft/dot-dom/tree/ic/smarter-state-perservation) branch, in order to enable nested dynamic components and lifecycle methods (still within 512 bytes :blush:), and the code has changed a bit. So base your investigations on the code from there.

**Comment by wavesoft** [comment]
Actually I went through and implemented this feature on `0.3.0` release. :+1:

**Comment by jhnns** [comment]
Wow, awesome! 👍 🎉

I'm really excited to try this out in my next web components project.
